service/src/main/java/bio/terra/pipelines/service/PipelineRunsService.java

```diff
@@ -93,21 +96,29 @@ public Map<String, Map<String, String>> preparePipelineRun(
       throw new InternalServerErrorException("%s workspaceId not defined".formatted(pipelineName));
     }
 
+    if (getPipelineRun(jobId, userId) != null) {
+      throw new BadRequestException("JobId %s already exists".formatted(jobId));
```

Comment:
should we put something here like "use the x endpoint to get details for it" or something?

Reply:
love it